### DISCUSSION TITLE
Multiple Data Providers in one form

### DISCUSSION THREAD

**Discussion by BillThyCat** [Budibase/budibase#6319] [Help] [answered]
I am trying to query data from the same database but I need to be able to query from 2 different tables.

We have job #'s assigned to all of our jobs, each job has phase and cost codes assigned to that job #.

I have an option picker currently working to search the job #, I then need to pass the job # on to the next area where they select the phase and cost code (which it needs the job # to display the unique phase/cost codes for that job.).  If I pull it all as one query it's over 300,000 records when there are only 2300 open jobs, each job has 5-500 phase/cost codes assigned to them.  It also tends to time out with the 10000 ms error when querying that much data.  I'm running the self hosted version and can't find where to change that setting, but I also don't really want it to take 10-15 seconds to display.  With my current query of 2300 jobs it's almost instant.

I have both data providers bound first, then the form is bound to the last one on the list, from there I can create two option pickers and select the data source for each one, the first one works exactly the way I want it.  The second one will only display the phase/cost codes for thatever job I had to bind to in order to create the query.

I have gone in and set the define action on both pickers to re-run the query with the formfield selected from the picker and even did a refresh data provider and it never properly pulls the correct data.  It pulls data, but again just whatever I bound to create the query.

Forgive me as I'm still new to Budibase and this will be the first app I've created in it.

The data is stored in a Microsoft SQL server and I only need read only access to it.

Thanks for any info you can provide, examples you can point me to, etc.

Below are my queries
List Jobs
SELECT [job_no]
      ,[job_name]
      ,[job_off_line_flg]
  FROM [StarBldr].[dbo].[JCJobs]
  Where job_off_line_flg <> 'A'

List Phase/Cost Codes
SELECT JCJobs.job_no
      ,[job_name]
      ,[job_off_line_flg]
      ,([phase_no] + '-' + [cost_no] + '-' + [cost_type] + '-' + [phase_desc]) as PCCode
  FROM [StarBldr].[dbo].[JCJobs],[StarBldr].[dbo].[JCCostMast]
  Where JCJobs.job_no = JCCostMast.job_no and JCJobs.job_no like {{jobnumber}}

**Comment by melohagan** [comment, marked as answer]
Hi @BillThyCat 

>I have gone in and set the define action on both pickers to re-run the query with the formfield selected from the picker and even did a refresh data provider and it never properly pulls the correct data. It pulls data, but again just whatever I bound to create the query.

Is that for an OnChange event (https://docs.budibase.com/docs/actions#on-change)
Make sure you are using the correct binding: `{{ Field Value }}`

Is there a reason you don't want to have two screens (one of which could be a modal:  https://docs.budibase.com/docs/screens#modal-screen)?

If you're having difficulties managing data and filtering on the single screen, having multiple screens and sharing data between them via <a href="https://docs.budibase.com/docs/url-parameters">URL Variables</a> may be easier.

**Reply by BillThyCat** [reply]
Ok, I have a better understanding on these.  Still finding a few that are kicking my butt.
